doc/common-issues.rst

```diff
@@ -0,0 +1,101 @@
+In addition, you can also upgrade to PyMongo >=3.9, which will remove the spurious error.
+
+
+MongoClient fails ConfigurationError: Unknown option type_registry
+------------------------------------------------------------------
+
+MongoClient has historically accepted CodecOptions's members as keyword arguments,
+however this does not work for type_registry in PyMongo <3.8:
+
+.. code-block::
+
+	>>> client = MongoClient(type_registry=TypeRegistry)
+	...
+	pymongo.errors.ConfigurationError: Unknown option type_registry
+
+To fix this, upgrade PyMongo to version 3.8 or above, or pass ``codec_options`` separately as
+its own keyword argument like so:
+
+.. code-block::
+	>>> client = MongoClient()
+	>>> coll = client.db.coll.with_options(codec_options=CodecOptions(type_registry=TypeRegistry()))
+
+
+DeprecationWarning: count is deprecated
+---------------------------------------
+
+PyMongo will no longer support :meth:`pymongo.cursor.count` in the near future.
+Instead, use :meth:`pymongo.collection.count_documents`.
+Note that this is NOT the same as ``Cursor.count_documents`` (which does not exist),
+this is a method of the Collection class, so you must call it on a collection object:
+
+.. code-block::
+
+	>>> client = MongoClient()
+	>>> num_documents = client.db.coll.count_documents({})
+
+
+Timeout when accessing MongoDB from PyMongo with tunneling
+----------------------------------------------------------
+
+When attempting to connect to a replica set MongoDB instance over an SSH tunnel
+will receive the following error:
+
+.. code-block::
+
+	File "/Library/Python/2.7/site-packages/pymongo/collection.py", line 1560, in count
+		return self._count(cmd, collation, session)
+	  File "/Library/Python/2.7/site-packages/pymongo/collection.py", line 1504, in _count
+		with self._socket_for_reads() as (sock_info, slave_ok):
+	  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/contextlib.py", line 17, in __enter__
+		return self.gen.next()
+	  File "/Library/Python/2.7/site-packages/pymongo/mongo_client.py", line 982, in _socket_for_reads
+		server = topology.select_server(read_preference)
+	  File "/Library/Python/2.7/site-packages/pymongo/topology.py", line 224, in select_server
+		address))
+	  File "/Library/Python/2.7/site-packages/pymongo/topology.py", line 183, in select_servers
+		selector, server_timeout, address)
+	  File "/Library/Python/2.7/site-packages/pymongo/topology.py", line 199, in _select_servers_loop
+		self._error_message(selector))
+	pymongo.errors.ServerSelectionTimeoutError: localhost:27017: timed out
+
+This is due to the fact that PyMongo discovers replica set members using the response from the isMaster command which
+then contains the address and ports of the other members. However, these addresses and ports will not be accessible through the SSH tunnel. Thus, this behavior is unsupported.
+You can, however, successfully connect to a single MongoDB node using SSH tunneling.
```

Comment:
"connect directly to a single MongoDB node using with the directConnection=True option with SSH tunneling."

Reply:
Done.